packages/amplify-codegen-appsync-model-plugin/src/__tests__/visitors/appsync-javascript-visitor.test.ts

```diff
@@ -100,7 +100,7 @@ export enum SimpleEnum {
 
 export declare class SimpleNonModelType {
   readonly id: string;
-  readonly names?: string[];
+  readonly names?: (string | null)[];
```

Comment:
I don't know if there's a unit test to make sure that: 

```graphql
foo: [Bar!]
```

Is generated correctly?

As in that case we would want the following generated since the elements themselves should be non-null, just the array nullable:

```ts
readyonly foo?: Bar[];
```

Reply:
@danrivett, I added a new Bar type to the tests and included it in the SimpleModel to cover this case.